src/Microsoft.Health.Fhir.SqlServer/Features/Search/SqlServerSearchService.cs

```diff
@@ -213,14 +228,15 @@ private async Task<SearchResult> SearchImpl(SearchOptions searchOptions, bool hi
                         if (matchCount >= searchOptions.MaxItemCount && isMatch)
                         {
                             moreResults = true;
-                            continue;
-                        }
 
-                        // See if this resource is a continuation token candidate and increase the count
-                        if (isMatch)
-                        {
-                            newContinuationId = resourceSurrogateId;
-                            matchCount++;
+                            // At this point we are at the last row.
+                            // if we have more columns, it means sort expressions were added.
+                            if (reader.FieldCount > 9)
```

Comment:
Be careful as a change has just gone in that added another field to the projection. Maybe it would be safer to use the column name after all.